### PR TITLE
Use GraphQLModules

### PR DESCRIPTION
**Pull request by ardatan**
- Use GraphQL-Modules instead of a factory function and add AccountsModule to support and also it exposes typeDefs, schema, resolvers for non-GraphQL-Modules project.
- Update documentation related to these changes

### CODE REVIEW

**Comment by pradel** [comment]
Hey, thanks for the pr, can you give a bit of context for these changes please?

**Comment by ardatan** [comment]
Hi @pradel,
Sorry for late response. 
We just released GraphQL-Modules which gives you to implement your GraphQL-Modules in modular approach together with features like Depenedency Injection, Resolvers Composition etc.
I am one of the contributors of that project.
You can read more in this article and its website;
https://medium.com/the-guild/graphql-modules-feature-based-graphql-modules-at-scale-2d7b2b0da6da
https://graphql-modules.com

We would love to see AccountsJS has built-in GraphQL-Modules support, and uses the benefits of our library. @dotansimha is one of the main contributors of GraphQL-Modules, and I think he also contributed to `graphql-api` package of yours.

We also want to write a blog post about a GraphQL boilerplate with Authentication using GraphQL-Modules and AccountsJS together with GraphQL Code Generator's TypeScript benefits that will be so much easy to start a project for beginners.

To sum up, the benefits are;
- You don't need to think about the support of GraphQL libraries and other stuff. This is what GraphQL-Modules will take care of it for you. 
- GraphQL-Modules has built-in support for Apollo Server and graphql-yoga even if they're not using GraphQL-Modules.
- GraphQL-Modules also supports GraphQL Code Generator out of the box which generates client-side and server-side typings including resolvers signatures, and strictly-typed React-Apollo and Apollo Angular components and services. That will be also included in your graphql-api package if you start to use GraphQL-Modules with this PR.
- https://medium.com/the-guild/graphql-code-generator-v0-11-15bb9b02899e

**Comment by dotansimha** [comment]
I think it's great :) 
It's exposing a clean `GraphQLModule` which you can use with `graphql-modules`, but the module is also independent, so you can get `typeDey`/`resolvers`/`GraphQLSchema` as standalone and use it without `graphql-modules`.

@davidyaha what do you think? 😄

**Comment by TimMikeladze** [comment]
I think this is a great idea, will review soon, thanks for the contribution!

File: packages/graphql-api/src/modules/accounts-password/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import TypesTypeDefs from './schema/types';
+import getQueryTypeDefs from './schema/query';
+import getMutationTypeDefs from './schema/mutation';
+import getSchemaDef from './schema/schema-def';
+import { Query } from './resolvers/query';
+import { Mutation } from './resolvers/mutation';
+import { IAccountsRequest } from '../accounts';
+import { mergeGraphQLSchemas } from '@graphql-modules/epoxy';
+
+export interface IAccountsPasswordModuleConfig {
```
Comment (pradel) [inline packages/graphql-api/src/modules/accounts-password/index.ts]: Can you remove the `I` in front of the types please? (to keep consistency with the code we already have))

**Comment by pradel** [comment]
I like the idea too!
Will take a deeper look when I have some time :)

File: packages/graphql-api/src/modules/accounts-password/index.ts (hunk 1 above)
Comment (ardatan) [reply packages/graphql-api/src/modules/accounts-password/index.ts]: Done :)

**Comment by pradel** [comment]
Code looks good and e2e tests are passing!
I just have one question since I am not familiar with graphql-modules.
How can a user extends the User type to add additional fields?
For example right now you can do it like this:
```
const typeDefs = `
  extends type User {
	firstname: String!
  }
`;
```

**Comment by pradel** [comment]
Same when we are creating a new user, how to extends the createUser input with new variables?

**Comment by ardatan** [comment]
Everything is same if you're not using GraphQL-Modules, the exported schema is exactly same like before; so you can still use `extend` for extending inputs and other types. But for GraphQL-Modules users, there is no need to add `extend` operator, because GraphQL-Modules handles that for you :) The only thing you need to do is to import `AccountsModule` to your application module which contains extra fields etc.
```ts
import { AccountsModule } from '@accounts/graphql-api`
import { GraphQLModule } from '@graphql-modules/core`

export const AppModule = new GraphQLModule({
  typeDefs: gql`
    type User {
	firstname: String!
  }
  `,
  resolvers: {...},
  imports: [
   AccountsModule.forRoot({...})
  ]
});

const { typeDefs } = AppModule;
// Now typeDefs has a User type with firstname and other fields from AccountsModule
```

File: packages/graphql-api/README.md [hunk 2]
```diff
@@ -46,19 +46,21 @@ const accountsServer = new AccountsServer({
...
+});
 ```
 
-Now, add `accountsGraphQL.typeDefs` to your schema definition (just before using it with `makeExecutableSchema`), and extend your resolvers object with `accountsGraphQL.resolvers`, for example:
+Now, add `accountsGraphQL.typeDefs` to your schema definition (just before using it with `makeExecutableSchema`), and merge your resolvers object with `accountsGraphQL.resolvers` by using `@graphql-tools/epoxy`, for example:
 
 ```js
 import { makeExecutableSchema } from 'graphql-tools';
-import { merge } from 'lodash';
+import { mergeGraphQLSchemas, mergeResolvers } from '@graphql-tools/epoxy;';
```
[the remark this answers is not in the document]
Comment (ardatan) [reply packages/graphql-api/README.md]: Thank you :)

**Comment by pradel** [comment]
Amazing thanks for the explanation @ardatan!
We will have to reflect that in the documentation :)

**Comment by ardatan** [comment]
@pradel Thank you for your good thoughts :) It is really great to see these two projects together.

**Comment by davidyaha** [comment]
@dotansimha Graphql Modules sounds great! Congrats on open sourcing it!!
Hadn't tried it myself yet but it looks like a very good approach and could be very much inline with accounts-js IMO.
Unfortunately I don't have the time to review but it seems like @pradel and @TimMikeladze are on it.
Thanks for the contribution @ardatan!

**Comment by dotansimha** [comment]
@pradel @TimMikeladze @davidyaha Any chance we can get it merged and released soon? 😄 
Thanks!

**Comment by TimMikeladze** [comment]
I can merge and cut a release within the next 24 hours.

**Comment by TimMikeladze** [comment]
Thanks for the PR 👍 

Merged and published. `0.5.0-alpha.1`
